### DISCUSSION TITLE
API body size limits

### DISCUSSION THREAD

**Discussion by crice88** [vercel/next.js#34450] [Help]
Here's my use case:

We have a stand alone Express API that our Nextjs app communicates with. One of our features is to allow users to download large files/folders. 

Currently, I have a Next API route that makes a request to our express server, the server makes a call to S3 to gather file information and we build up the folder structure as a stream using `archiver`. 

We then stream the response back to our Next API route which then streams it back to the browser. This functionality is kicked off using a link with the href pointing to our Next API route.

Our Next API handler ('/api/download')
```
const handler = async (req: NextIronRequest, res: NextApiResponse) => {
  const url = EXTERNAL_API_ROUTE;

  const response = await fetch(url);

  if (!response.ok) throw new Error(`unexpected response ${response.statusText}`);

  res.setHeader('Content-Type', 'application/zip');
  res.setHeader('Content-Disposition', 'attachment; filename=foo.zip');

  res.send(response.body);
};
```

Usage
```
<a href='/api/download'>Download Me!</a>
```

Currently, I am getting a warning in development that the `API response for /api/download exceeds 4MB`. It seems silly to me that this would be removed in a later release as I'd think this could be a common use case.

If it will be removed later, can someone think of an alternative?

**Comment by 2stash** [comment]
@crice88 Hi again, just answered you on: https://github.com/vercel/next.js/discussions/28512

Seems like this is an intended issue, and also seems like there is no intention of changing it.

See recent opened and closed issue: https://github.com/vercel/next.js/issues/33162

Documentation for 4mb limit: https://nextjs.org/docs/messages/api-routes-body-size-limit

I think the only recourse is to maybe open another issue and try to get it upvoted.
